patches/net/minecraft/server/level/ChunkMap.java.patch

```diff
@@ -97,6 +97,27 @@
              EntityType<?> entitytype = p_140200_.getType();
              int i = entitytype.clientTrackingRange() * 16;
              if (i != 0) {
+@@ -1260,6 +_,20 @@
+         });
+     }
+ 
++    // Neo: Getter for players watching an entity.
++    public List<ServerPlayer> getPlayersWatching(Entity entity) {
++        var trackedEntity = entityMap.get(entity.getId());
++        if (trackedEntity != null) {
++            var ret = new java.util.ArrayList<ServerPlayer>(trackedEntity.seenBy.size());
++            for (var connection : trackedEntity.seenBy) {
++                ret.add(connection.getPlayer());
++            }
++            return java.util.Collections.unmodifiableList(ret);
++        } else {
++            return List.of();
++        }
++    }
++
```

Comment:
Curiosity: Why not in an extension? Granted patching wise it does not matter, just wanted to know the reasoning.

Reply:
Okiedokie